### PR TITLE
Upgrade ListObjects to ListObjectsV2

### PR DESCRIPTION
**Pull request by Pisush**
`ListObjectsV2` is the revised List Objects API and is recommend for new application development.

https://docs.aws.amazon.com/sdk-for-go/api/service/s3/#S3.ListObjectsV2

### CODE REVIEW

File: s3/container.go [hunk 1]
```diff
@@ -79,7 +79,7 @@ func (c *container) Items(prefix, cursor string, count int) ([]stow.Item, string
 	// and correctly return the next list of items in subsequent requests.
 	marker := ""
 	if *response.IsTruncated {
-		marker = containerItems[len(containerItems)-1].Name()
+		marker = containerItems[len(containerItems)-2].Name()
```
Comment (Pisush) [inline s3/container.go]: `Marker` starts from the current item
`StartAfter` starts from the one after this item

File: s3/container.go [hunk 2]
```diff
@@ -77,12 +77,12 @@ func (c *container) Items(prefix, cursor string, count int) ([]stow.Item, string
 	// If not, provide the file name of the last item as the next marker. S3 lists
 	// its items (S3 Objects) in alphabetical order, so it will receive the item name
 	// and correctly return the next list of items in subsequent requests.
-	marker := ""
+	startAfter := ""
 	if *response.IsTruncated {
-		marker = containerItems[len(containerItems)-1].Name()
+		startAfter = containerItems[len(containerItems)-2].Name()
```
Comment (Pisush) [inline s3/container.go]: `Marker` starts from the current item
`StartAfter` starts from the one after this item

File: s3/container.go [hunk 3]
```diff
@@ -77,12 +80,12 @@ func (c *container) Items(prefix, cursor string, count int) ([]stow.Item, string
...
```
Comment (naysayer) [inline s3/container.go]: Thinking we should preform a check here to make sure that len-2 is okay. We may get an out of bounds error I want to protect against.

Comment (naysayer) [reply s3/container.go]: Then return an error in the event it cannot be accessed.

**Pull request merged**